test/extensions/filters/http/common/fuzz/filter_fuzz_test.cc

```diff
@@ -0,0 +1,159 @@
+      filter->decodeTrailers(trailers);
+    }
+  }
+
+  // This creates the filter config and runs decode.
+  void fuzz(const envoy::extensions::filters::network::http_connection_manager::v3::HttpFilter&
+                proto_config,
+            const test::fuzz::HttpData& data) {
+    try {
+      // Try to create the filter. Exit early if the config is invalid or violates PGV constraints.
+      ENVOY_LOG_MISC(info, "filter name {}", proto_config.name());
+      auto& factory = Config::Utility::getAndCheckFactoryByName<
+          Server::Configuration::NamedHttpFilterConfigFactory>(proto_config.name());
+      ProtobufTypes::MessagePtr message = Config::Utility::translateToFactoryConfig(
+          proto_config, factory_context_.messageValidationVisitor(), factory);
+      cb_ = factory.createFilterFactoryFromProto(*message, "stats", factory_context_);
+      cb_(filter_callback_);
+    } catch (const EnvoyException& e) {
+      ENVOY_LOG_MISC(debug, "Controlled exception {}", e.what());
+      return;
+    }
+
+    decode(filter_.get(), data);
+    reset();
+  }
+
+  void reset() {
+    if (filter_.get() != nullptr) {
+      filter_.get()->onDestroy();
+    }
+    filter_.reset();
+  }
+
+  NiceMock<Server::Configuration::MockFactoryContext> factory_context_;
+  NiceMock<Http::MockStreamDecoderFilterCallbacks> callbacks_;
+  NiceMock<Http::MockFilterChainFactoryCallbacks> filter_callback_;
+  std::shared_ptr<Http::StreamDecoderFilter> filter_;
+  Http::FilterFactoryCb cb_;
+  NiceMock<Envoy::Network::MockConnection> connection_;
+  Network::Address::InstanceConstSharedPtr addr_;
+};
+
+DEFINE_PROTO_FUZZER(const test::extensions::filters::http::FilterFuzzTestCase& input) {
+  static PostProcessorRegistration reg = {[](test::extensions::filters::http::FilterFuzzTestCase*
+                                                 input,
+                                             unsigned int seed) {
+    // This ensures that the mutated configs all have valid filter names and type_urls.
+    static const std::vector<absl::string_view> filter_names = Registry::FactoryRegistry<
+        Server::Configuration::NamedHttpFilterConfigFactory>::registeredNames();
+    static const auto factories =
+        Registry::FactoryRegistry<Server::Configuration::NamedHttpFilterConfigFactory>::factories();
+    if (std::find(filter_names.begin(), filter_names.end(), input->config().name()) ==
+        std::end(filter_names)) {
+      absl::string_view filter_name = filter_names[seed % filter_names.size()];
+      input->mutable_config()->set_name(std::string(filter_name));
+    }
+    // Set appropriate type_url.
+    auto& factory = factories.at(input->config().name());
+    input->mutable_config()->mutable_typed_config()->set_type_url(
+        factory->createEmptyConfigProto()->GetDescriptor()->full_name());
+    // Scrubs away any use of hidden_envoy_deprecated_* for the filter config.
```

Comment:
Is this only a problem on the compile time options build where deprecated fields are fatal? It seems that in general, we can fuzz these similar to normal fields.

Reply:
The issue with fuzzing them as normal is that they almost never reach good coverage strings since LPM will fuzz them as a `google.protobuf.Struct`. The less it sees with good coverage, the more the depth of the struct increases. (Rather than a string populated depth-constrainted `google.protobuf.Any`)

Reply:
OK, so it doesn't have to do with the hidden deprecated aspect but the `Struct` structure vs. dictionary fuzzing. Got it.